go-controller/pkg/ovn/secondary_localnet_network_controller.go

```diff
@@ -360,11 +403,82 @@ func (oc *SecondaryLocalnetNetworkController) newRetryFramework(
 }
 
 func (oc *SecondaryLocalnetNetworkController) localnetPortNetworkNameOptions() map[string]string {
-	localnetLSPOptions := map[string]string{
-		"network_name": oc.GetNetworkName(),
+	return map[string]string{
+		"network_name": oc.findNetworkName(),
+	}
+}
+
+func (oc *SecondaryLocalnetNetworkController) checkBridgeMapping(obj interface{}) error {
+	if !config.OVNKubernetesFeature.EnableInterconnect {
+		return nil
+	}
+
+	pod, isPod := obj.(*corev1.Pod)
+	if !isPod {
+		return fmt.Errorf("failed casting %T object to *knet.Pod", obj)
+	}
+
+	if !util.PodScheduled(pod) {
+		return nil
+	}
+
+	if oc.controllerNodeName != pod.Spec.NodeName {
+		return nil
+	}
+
+	isNADKnown, err := oc.isPodAttachedToKnownNAD(pod)
+	if err != nil {
+		return err
+	}
+	if !isNADKnown {
+		return nil
```

Comment:
What happens if the NAD becomes known and our business logic should apply? We don't reconcile this. I guess not a major issue if we are just reporting an event...until we do something more and then it becomes an issue.
I guess I am just interested if this was considered for now.